packages/polytechcolloscopeclient/lib/src/polytechcolloscopeclient_base.dart

```diff
@@ -0,0 +1,110 @@
+import 'package:beautiful_soup_dart/beautiful_soup.dart';
+import 'package:html/dom.dart';
+import 'package:requests_plus/requests_plus.dart';
+
+import 'colloscope_datastructs.dart';
+import 'consts.dart';
+
+class PolytechColloscopeClient {
+  final String username, password;
+
+  PolytechColloscopeClient(this.username, this.password) {
+    assert(username.isNotEmpty);
+    assert(password.isNotEmpty);
+  }
+
+  Future<List<Student>> fetchStudents(Year year) async {
+    var page = await RequestsPlus.get(Consts.kholleURL[year]!,
+        userName: username, password: password);
+
+    BeautifulSoup bs = BeautifulSoup(page.body);
+
+    //This isolates the student list part of the page, it consists in a list of <li> elements like
+    // <li><a href="?page=colles_2A&amp;id_etudiant=STUDENT_ID">INITAL. SUNRNAME</a> (groupe GROUPNUMBER)</li>
+    var studentsList = bs.find("table.colles")?.find("td > ul");
```

Comment:
Replace the ? with a ! to avoid writing ? everywhere.
And by the way, it avoids making the computer do null checks.

Reply:
Your change doesn't work but adding an eclamation mark at the end of the line does